stdlib/http/src/main/ballerina/http/auth/authn_filter.bal

```diff
@@ -54,25 +54,49 @@ public type AuthnFilter object {
     }
 };
 
-function handleAuthnRequest(AuthnHandler[] authnHandlers, Request request) returns boolean|error {
+function handleAuthnRequest(AuthnHandler?[]|AuthnHandler?[][] authnHandlers, Request request) returns boolean|error {
+    if (authnHandlers is AuthnHandler?[]) {
+        return checkForAuthnHandlers(authnHandlers, request);
+    } else {
+        (boolean|error?)[] responseArray = [];
+        int count = 0;
+        foreach AuthnHandler?[] authnHandler in authnHandlers {
+            responseArray[count] = checkForAuthnHandlers(authnHandler, request);
```

Comment:
Can we break early, without us having to go through all the authentication. 

Since we are evaluating AND in this situation (array of arrays),  if any of the group was evaluated to be "false" there is no need to proceed invoking other authenticators.

Reply:
Yeah. Could be improved without the use of the `responseArray`. Done with https://github.com/ballerina-platform/ballerina-lang/pull/15363/commits/3ed53508dbebdbee9be2b86662d73ac4a81e5446